### PR TITLE
feat: add setCookie utility

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- add `setCookie`
  - since we have `getCookieObject` implementation it was logical to implement our own setter utility
- on the server we should use native req.cookies
- base for implementation https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie
  - also looked at other implementations like https://www.npmjs.com/package/cookie which are mostly bloated but did include same edge cases I included in our utility
- refactor usages of `document.cookie` to use `setCookie`
  - tested cookies were set correctly when reading on client and server (in boot)

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1432 #done

### CODE REVIEW

File: packages/shared/src/hooks/useCookieBanner.ts
Comment (capJavert) [inline packages/shared/src/hooks/useCookieBanner.ts:21]: I think we had a bug in our code where our cookie banner `ilikecookies` cookie was always a session cookie. 

This was due to `expires` being set as `number` of seconds and per spec it should be date formatted string. 

I replaced it with `max-age` which correctly sets expiration of cookie in seconds (in this case 10 years). 

We did not notice this because nowadays browsers do not clear session cookies until their process is closed in full vs closing a tab.

**Review by rebelchris — APPROVED** [review]
Thanks for picking this up, looks neat.
Just maybe wait for approval from Lee just in case 🙏

**Comment by capJavert** [comment]
@sshanzel could you also take a look if you have time?

File: packages/shared/src/lib/cookie.ts
Comment (sshanzel) [inline packages/shared/src/lib/cookie.ts:71]: I think we can use the `isNullOrUndefined`.

File: packages/shared/src/hooks/useCookieBanner.ts
Comment (sshanzel) [reply packages/shared/src/hooks/useCookieBanner.ts:21]: Amazing. Thanks for the retro fix. I think we can introduce a variable named `const TEN_YEARS` or something similar, to make easier to understand in one look.

File: packages/shared/src/lib/cookie.ts
Comment (sshanzel) [inline packages/shared/src/lib/cookie.ts:50]: Just some non-blocking cleanups.
```suggestion
interface CookieOptions {
  domain: string;
  expires: Date | string;
  maxAge: number;
  partitioned: boolean;
  path: string;
  sameSite: 'lax' | 'strict' | 'none';
  secure: boolean;
}

export const setCookie = (
  name: string,
  value: string | number | boolean,
  options: Partial<CookieOptions> = {},
): void => {
```

File: packages/webapp/pages/join.tsx
Comment (sshanzel) [inline packages/webapp/pages/join.tsx:69]: Also this, I think we can set a variable somewhere.

**Review by sshanzel — APPROVED** [review]
Wow, great work. Amazing to see you added the test cases too.

Just had non-blocking comments. Other than that, everything looks great.

File: packages/shared/src/hooks/useCookieBanner.ts
Comment (capJavert) [reply packages/shared/src/hooks/useCookieBanner.ts:21]: @rebelchris @sshanzel one fun fact for Chrome https://developer.chrome.com/blog/cookie-max-age-expires/#:~:text=With%20this%20change%2C%20Chrome%20caps,set%20to%20400%20days%20instead

I think we can still leave 10 years for other browsers but something to keep in mind.

**Pull request merged**
